l10n_it_fatturapa_out/wizard/efattura.py

```diff
@@ -145,12 +145,20 @@ def format_quantity(line):
             res = ("{qta:.{precision}f}".format(qta=quantity, precision=uom_precision),)
             return res[0]
 
-        def get_vat_number(vat):
-            # return vat[2:].replace(' ', '') if vat else ""
-            return vat[2:] if vat else ""
-
-        def get_vat_country(vat):
-            return vat[:2].upper() if vat else ""
+        def get_id_fiscale_iva(partner):
+            id_paese = partner.country_id.code
+
+            if partner.vat:
+                if partner.country_id.code == "IT" and partner.vat.startswith("IT"):
+                    id_codice = partner.vat[2:]
```

Comment:
Nel caso in cui stiamo compilando il nodo `IdTrasmittente/IdCodice`, qui potrebbe andarci il codice fiscale, attualmente gestisci questo caso nel XML; che ne pensi di gestirlo invece in questo punto?
Credo che il codice Python sia più facile da gestire rispetto al XML, ad esempio nel caso si vogliano cambiare le logiche con override/inheriting.

Reply:
Fatto. Ho dovuto passare un parametro ulteriore alla funzione, che non sapeva se veniva chiamata per `IdTrasmittente` o da altri punti del template...